lib/teleterm/services/userpreferences/userpreferences.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package userpreferences
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	userpreferencesv1 "github.com/gravitational/teleport/api/gen/proto/go/userpreferences/v1"
+	api "github.com/gravitational/teleport/gen/proto/go/teleport/lib/teleterm/v1"
+)
+
+func Get(ctx context.Context, rootClient Client, leafClient Client) (*api.UserPreferences, error) {
+	rootPreferencesResponse, err := rootClient.GetUserPreferences(ctx, &userpreferencesv1.GetUserPreferencesRequest{})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	rootPreferences := rootPreferencesResponse.GetPreferences()
+	clusterPreferences := rootPreferences.GetClusterPreferences()
+
+	if leafClient != nil {
+		preferences, err := leafClient.GetUserPreferences(ctx, &userpreferencesv1.GetUserPreferencesRequest{})
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
```

Comment:
this conditional should only overwrite cluster prefs and not userprefs.  preferences held in `UserPreferences` (from the root) are the settings we don't want to change like selected theme and what not

Reply:
Agree, but I think we are only overwriting the cluster preferences, no?

Reply:
Oh yes, my bad. Read it wrong